modules/_labs/select/react/lib/SelectBase.tsx

```diff
@@ -436,7 +436,7 @@ export default class SelectBase extends React.Component<SelectBaseProps> {
           <SelectMenu
             aria-activedescendant={options[focusedOptionIndex].id}
             aria-labelledby={ariaLabelledBy}
-            aria-required={ariaRequired || required}
+            aria-required={ariaRequired || required ? true : undefined}
```

Comment:
Nice. I was looking to make sure these properties were `undefined` or `true`. The recommendation is to never set to `false`

Reply:
ah, just curious why never false vs undefined?

Reply:
I'm not sure this is what Nicholas was referring to, but in this case I didn't want `aria-required="false"` to be unnecessarily applied to the `<ul>` rendered by `SelectMenu` (more concise to just omit `aria-required` entirely). Given [the way logical operators work in JavaScript](https://mariusschulz.com/articles/the-and-and-or-operators-in-javascript#using-logical-operators-with-non-boolean-values),  `ariaRequired || required` resolves to `false` if `ariaRequired` is falsy and `required` is explicitly set to `false`.

Therefore, if `ariaRequired` is `undefined` and `required` is `false`, `<SelectMenu aria-required={ariaRequired || required}>` resolves to `<SelectMenu aria-required={false}>`, resulting in `<ul aria-required="false">`. In order to get rid of the unnecessary `aria-required="false"`, I forced the `aria-required` prop to be set to `true` or `undefined` (but never `false`).

Note there are [certain attributes](https://stackoverflow.com/a/31164090) React will omit if they're set to `false` (or any other falsy value), but `aria-required` isn't one of them.

Reply:
Right. Undefined removes the attr altogether, which is what you'd want here. Nice work!

Reply:
I wasn't able to find concrete documentation on omitting attributes vs explicit `attribute="false"`. That's a recommendation from accessibility. I think some screen readers get confused and treat `attribute="false"` as a `true` value. Many attributes don't technically have a value, just a presence (like `<input required>`. Attributes can only be strings and all strings resolve as "truthy" in JavaScript.

I think it is just safer to omit attributes for the false case and have `="true"` for the true case unless the specification explicitly says to set it to `"false"` and the accessibility team agrees.